### PR TITLE
Update spectral_corner_frequency_from_surface_xdmf.py

### PR DESCRIPTION
**Pull request by NicoSchlw**
Fix eventDuration for constantly slipping faults and add a minimum frequency depending on the length of the time series

### CODE REVIEW

File: postprocessing/science/spectral_corner_frequency_from_surface_xdmf.py [hunk 1]
```diff
@@ -243,7 +244,15 @@ def ApproximateEventDurationAndHypocenter(faultxdmf, faultxyz):
         ASl = LoadSingleComponentParallel("ASl", surfaceData=False)
     ASR = np.abs(np.gradient(ASl, dtFault, axis=1))
     slippingElement = np.where(np.amax(ASR, axis=0) > slipRateThreshold, 1, 0)
-    duration = slippingElement[np.argmax(slippingElement):-np.argmax(np.flip(slippingElement))].size * dtFault
+    if np.all(slippingElement == 1):
+        print("Warning: at least one element is constantly slipping")
```
Comment (Thomas-Ulrich) [inline postprocessing/science/spectral_corner_frequency_from_surface_xdmf.py:248]: you mean 'there is always one element slipping during the whole simulation'?
(not necessarily always the same?).

File: postprocessing/science/spectral_corner_frequency_from_surface_xdmf.py [hunk 2]
```diff
@@ -243,7 +244,15 @@ def ApproximateEventDurationAndHypocenter(faultxdmf, faultxyz):
...
+        print("eventDuration is set to the entire length of the time series")
+        duration = (slippingElement.size - 1) * dtFault
+    elif np.argmax(np.flip(slippingElement)) == 0:
+        print("Warning: at least one element is still slipping at the last time step")
+        duration = slippingElement[np.argmax(slippingElement):-1].size * dtFault
+    else:
+        duration = slippingElement[np.argmax(slippingElement):-np.argmax(np.flip(slippingElement))].size * dtFault
```
Comment (Thomas-Ulrich) [inline postprocessing/science/spectral_corner_frequency_from_surface_xdmf.py:255]: why is this formula not valid for all cases?

**Review by Thomas-Ulrich — APPROVED** [review]

File: postprocessing/science/spectral_corner_frequency_from_surface_xdmf.py (hunk 2 above)
Comment (NicoSchlw) [reply postprocessing/science/spectral_corner_frequency_from_surface_xdmf.py:255]: When there is still an element slipping at the last time step `-np.argmax(np.flip(slippingElement))` is zero, which means duration is also zero

File: postprocessing/science/spectral_corner_frequency_from_surface_xdmf.py (hunk 1 above)
Comment (NicoSchlw) [reply postprocessing/science/spectral_corner_frequency_from_surface_xdmf.py:248]: Yes, `"Warning: at least one element is always slipping"` is probably better phrased

File: postprocessing/science/spectral_corner_frequency_from_surface_xdmf.py (hunk 2 above)
Comment (Thomas-Ulrich) [reply postprocessing/science/spectral_corner_frequency_from_surface_xdmf.py:255]: What about np.trim_zeros(slippingElement).size?

Comment (NicoSchlw) [reply postprocessing/science/spectral_corner_frequency_from_surface_xdmf.py]: Good idea, I changed it to np.trim_zeros

**Pull request merged**